openassessment/xblock/static/js/src/lms/oa_base.js

```diff
@@ -73,6 +74,25 @@ OpenAssessment.BaseView.prototype = {
     },
 
     /**
+     * Install click handlers to set usage key of an XBlock.
+     */
+    setUpUsageID: function() {
+        $(".action--submit").on("click", function(event) {
+            this.usageID = $(event.target).closest('.xblock-student_view-openassessment').data('usage-id');
+        });
+    },
+
+    /**
+     * Get usage key of an XBlock.
+     */
+    getUsageID: function() {
+        if (!this.usageID) {
+            this.usageID = $(this.element).data('usage-id');
```

Comment:
I'm confused by the two different ways to set ``this.usageID``. Why do you need the click handler if this method can just determine the correct usage ID on demand?